File: leetcode/1039-find-the-town-judge/find-the-town-judge.cpp
class Solution {
public:
    int findJudge(int n, vector<vector<int>>& trust) {
      vector<int> from(n+1);
      vector<int> to(n+1);
      for(int i=0;i<trust.size();i++){
          from[trust[i][0]]++;
          to[trust[i][1]]++;
        
      }
      for(int i=1;i<=n;i++){
          if(from[i]==0 && to[i]==n-1){
              return i;
          }
          
      }
      return -1;
    }
};

File: leetcode/1039-find-the-town-judge/find-the-town-judge.py
class Solution:
    def findJudge(self, n: int, trust: List[List[int]]) -> int:
        numTrust = [0] * (n + 1)

        # Increase trust value of trustee and decrease trust value of truster for each pair
        for person1, person2 in trust:
            numTrust[person1] -= 1
            numTrust[person2] += 1
        
        # Check if anyone achieves n-1 trust. This person is the town judge
        for i in range(1, len(numTrust)):
            if numTrust[i] == n - 1:
                return i
        
        # If no one achieves n-1 trust, then there is no town judge
        return -1
            
            
        